spec/Failsafe.spec.ts

```diff
@@ -171,6 +171,126 @@ describe(`Failsafe`, function() {
             expect(logger.stdout()).to.include(`[check-ansi-support] Colors not supported`);
         });
     });
+
+    describe('Pass arguments', () => {
+
+        it(`fails on unknown arguments`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run([`print-args`, '--foo=bar', '--bar=foo']);
+            expect(exitCode).to.equal(General_Failure, `Expected exit code of ${General_Failure}${ format(logger) }`);
+
+            expect(logger.stderr()).to.include(`Unknown argument '--foo=bar'`);
+        });
+
+        it(`passes specific arguments`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run([`print-args`, '[--foo]', '--foo=bar']);
+            expect(exitCode).to.equal(Success, `Expected exit code of ${Success}${ format(logger) }`);
+
+            expect(logger.stdout()).to.include([
+                `[print-args] Listing 1 arguments`,
+                `[print-args] --foo=bar`,
+                `[failsafe] Script 'print-args' exited with code 0`,
+            ].join('\n'));
+        });
+
+        it(`passes specific arguments interleaved`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run(['print-args[--foo][--bar]', '--foo', '--bar']);
```

Comment:
Yes, I was thinking about the same. But I would prefer ‘|’ over ‘,’ but we can support both…

Reply:
I'm curious, would you use `|` because of aesthetics, or is there a practical reason?

```
print-args[--foo|--bar|--baz|-p|-vvv|...]
print-args[--foo,--bar,--baz,-p,-vvv,...]
```

Reply:
Pure aesthetics. But I just realized that `|` will not work in that context as it would be interpreted…  `,` will do it.